### PR TITLE
Add return promise custom type

### PR DESCRIPTION
**Pull request by cacheung**
Add return promise custom type

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/jest/setup.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ jest.doMock('react-native', () => {
                 },
                 AEPEdge: {
                     extensionVersion: jest.fn(() => new Promise(resolve => resolve(''))),
-                    sendEvent: jest.fn(() => new Promise(resolve => resolve(null))),
+                    sendEvent: jest.fn(() => new Promise(resolve => resolve(''))),
```
Comment (cacheung) [inline tests/jest/setup.ts:31]: I need to change this to string, then the tests pass.

File: apps/AEPSampleApp/extensions/ConsentView.js [hunk 2]
```diff
@@ -63,4 +62,30 @@ function setDefaultConsent(allowed: boolean) {
   var collectConsentStatus = allowed ? {"val": "y"} : {"val": "n"};
   var defaultConsents: {[keys: string]: any} = {"consent.default": {"consents" : {"collect" : collectConsentStatus}}};
   MobileCore.updateConfiguration(defaultConsents);
-}
\ No newline at end of file
+}
```
Comment (cacheung) [inline apps/AEPSampleApp/extensions/ConsentView.js:65]: revert the style change in Edge branch. Will do it in a Consent PR.

File: tests/jest/setup.ts (hunk 1 above)
Comment (emdobrin) [reply tests/jest/setup.ts:31]: We should check for null before the for loop, and I think that would fix your tests too https://github.com/adobe/aepsdk-react-native/pull/114/files#diff-e010278071c5f1853863f926b699e59cdfbb76d4d01afa67b835f7470b4a5cccR61

Comment (cacheung) [reply tests/jest/setup.ts]: yes, works now.

File: packages/edge/js/Edge.js [hunk 3]
```diff
@@ -39,6 +39,32 @@ module.exports = {
...
+ 
+    let edgeEventArray = [];
+
+    if (!eventArray) {
+      return [];
+    }
+
+    for (let prop of eventArray) {
+      edgeEventArray.push(new EdgeEventHandle(prop.type, prop.payload));
+    }
```
Comment (emdobrin) [inline packages/edge/js/Edge.js:67]: For L61 -> 67 can we have something like:
if (eventArray) {
for (let prop of eventArray) {
      edgeEventArray.push(new EdgeEventHandle(prop.type, prop.payload));
    }
}

File: apps/AEPSampleApp/extensions/ConsentView.js (hunk 2 above)
Comment (emdobrin) [reply apps/AEPSampleApp/extensions/ConsentView.js:65]: should we revert this change?

Comment (cacheung) [reply apps/AEPSampleApp/extensions/ConsentView.js:65]: This is reverting the change for Consent View. Having all the styles back.  
I removed them from Consent PR.

**Review by emdobrin — APPROVED** [review]

**Pull request merged**